node/notifier/listeners.go

```diff
@@ -103,10 +110,13 @@ func (li *ledgerChannelListeners) createNewListener() <-chan query.LedgerChannel
 
 // getOrCreateListener returns the first listener, creating one if none exist.
 func (li *ledgerChannelListeners) getOrCreateListener() <-chan query.LedgerChannelInfo {
+	li.listenersLock.Lock()
```

Comment:
Using `defer li.listenersLock.Lock()` seems more efficient because you only have to use it once instead of calling `li.listenersLock.Unlock()` before each `return` statement

Reply:
This is actually what I tried first, but I ran into a deadlock. 

Because we use an inline `return` statement which calls into another function, it may not be clear the order in which the deferred mutex unlock and the function call happen. 

It's tempting to think the defer statement will be inserted "above" the entire `return` line, but that is not the case:

https://go.dev/play/p/PY1RaOR4q6R

(The subfunction call is evaluated before the defer statement is triggered).

In order to prevent a deadlock, we need to unlock the mutex before calling in to the other function (which needs to be able to lock the mutex). So `defer` doesn't seem so useful here.